src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,445 @@
+   */
+  @Prop() styleBorderColor = "rgb(0, 92, 129)";
+
+  /**
+   * This property allows you to select the gauge type. _(Circle or Line)_.
+   * Default is linear type.
+   */
+  @Prop() gaugeType: "Line" | "Circle" = "Line";
+
+  /**
+   *  Allows display current value. Default is disabled.
+   *
+   */
+  @Prop() showValue = false;
+
+  /**
+   * The Minimum Value of the gauge
+   *
+   */
+  @Prop() minValue: number;
+
+  /**
+   * The Currecnt value indicanding in the gauge
+   *
+   */
+  @Prop() currentValue: number = this.minValue;
+
+  /**
+   * This allows specifying the width of the circumference _(When gauge is Circle type)_ and the width of the bar _(When gauge is Line type)_ in % relative the component size.
+   *
+   */
+  @Prop() thickness = 10;
+
+  @State() maxValue: number;
+
+  @State() totValues = this.minValue;
+
+  @State() children = [];
+
+  @State() minorSize: number;
+
+  @Listen("gxGaugeRangeDidLoad")
+  onGaugeRangeDidLoad({ detail: childRange }) {
+    this.children = [...this.children, childRange];
+    this.totValues += childRange.amount;
+    childRange.element.addEventListener("gxGaugeRangeDidUnload", () => {
+      const inDex = this.children.findIndex(x => x === childRange);
+      this.children.splice(inDex, 1);
+      this.totValues -= childRange.amount;
+    });
+    childRange.element.addEventListener("gxGaugeRangeDidUpdate", () => {
+      const inDex = this.children.findIndex(x => x === childRange);
+      this.children.splice(inDex, 1, childRange);
+      this.totValues = 0;
+      for (const childInstance of this.children) {
+        this.totValues += childInstance.amount;
+      }
+    });
+  }
+
+  componentWillLoad() {
+    this.totValues = 0;
+  }
+
+  private calcThickness() {
+    return typeof this.thickness === "number" &&
+      (this.thickness > 0 && this.thickness <= 100)
+      ? this.thickness / 5
+      : 10;
+  }
+
+  ////////////////////////////////////////
+
+  render() {
+    this.minorSize =
+      this.element.offsetHeight > this.element.offsetWidth
+        ? this.element.offsetWidth
+        : this.element.offsetHeight;
+    function calcPercentage(comp) {
+      return (
+        (comp.currentValue - comp.minValue) *
+        100 /
+        (comp.maxValue - comp.minValue)
+      );
+    }
+    const childRanges = Array.from(
+      this.element.querySelectorAll("gx-gauge-range")
+    );
+    this.maxValue = this.totValues + this.minValue;
+    this.totValues = this.maxValue - this.minValue;
+    if (this.gaugeType === "Circle") {
+      const svgRanges = [];
+      let acumul = 0;
+      //////////////////////////////////////////
+      function calcPositionRange(preValue, value) {
+        acumul += preValue;
+        return value + acumul;
+      }
+      //////////////////////////////////////////
+      for (let i = childRanges.length - 1; i >= 0; i--) {
+        svgRanges.push(
+          <circle
```

Comment:
The logic that creates a `<circle>` element could be extracted to a new function to help make the code more compact and readable.